crates/factor-key-value-sqlite/src/lib.rs

```diff
@@ -0,0 +1,72 @@
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use anyhow::Context;
+use serde::{Deserialize, Serialize};
+use spin_factor_key_value::MakeKeyValueStore;
+use spin_key_value_sqlite::{DatabaseLocation, KeyValueSqlite};
+
+pub struct SpinKeyValueStore {
+    base_path: PathBuf,
+}
+
+impl SpinKeyValueStore {
+    pub fn new(base_path: Option<PathBuf>) -> anyhow::Result<Self> {
```

Comment:
Let's make sure to doc this stuff as well (e.g., that if `base_path` is `None`, current_dir will be used.